rfc/rfc018-openid-credential-issuance.md

```diff
@@ -0,0 +1,140 @@
+"Verifiable Credential", "Credential Subject", "Holder" and "Issuer" as defined by the [Verifiable Credential Data Model](https://www.w3.org/TR/vc-data-model/), and
+"Wallet", "Client Metadata" and "Credential Offer" as defined by [OpenID 4 Verifiable Credential Issuance](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html).
+
+## 3. Overview
+
+While OpenID4VCI mainly describes issuing credentials through a browser or to a mobile wallet app,
+this specification describes how to use OpenID4VCI to issue credentials server-to-server, without user interaction from the holder.
+
+Nodes MUST support both processes of issuing and receiving credentials over OpenID4VCI.
+
+## 3.1 Supported Flows and Features
+
+Conforming implementations MUST support the following features:
+
+- Pre-authorized code flow
+- Credential offer:
+  - with `credential_offer` parameter
+  - with `string` values, referring to a credential's `id` in `credentials_supported`
+- Credentials in JSON-LD format
+- Issuer- and wallet metadata
+
+## 3.2 Credential Offer Response
+
+The OpenID4VCI specification does not specify responses for the credential offer.
+But since this specification describes server-to-server interaction
+a response is useful for the issuer to know whether the credential offer was accepted.
+
+### 3.2.1 Credential Offer Success Response
+
+In case the wallet considers the offer valid, it MUST respond with HTTP status `200 OK`.
+This is regardless of whether the wallet will actually revrieve the wallet.
+
+If the wallet decides to retrieve the credential synchronously, it SHOULD respond with `status` set to `credential_received`.
+Below is a non-normative example of such a response:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "status": "credential_received"
+}
+```
+
+### 3.2.2 Credential Offer Error Response
+
+This specification extends the specification with the following error responses a wallet MUST support when receiving a credential offer:
+
+- `invalid_request`: the offer is missing or invalid.
+- `invalid_grant`: the offer does not contain a grant the wallet could use to retrieve the credential.
+
+It MUST respond with HTTP status `400 Bad Request` and a response body containing the error code.
+
+Below is a non-normative example of such a response:
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "error": "invalid_grant"
+}
+```
+
+## 4. Client Metadata Discovery using DID documents
+
+To offer a credential the issuer needs to discover the wallet's credential offer endpoint, which is registered in the client metadata.
+
+To allow a (Nuts) wallet to be discovered, the wallet MUST be represented by a resolvable DID.
+The DID document MUST contain a `node-http-services-baseurl` service, which contains the base URL for node-to-node HTTP services.
```

Comment:
```suggestion
The DID document MUST contain a `node-http-services-baseurl` service type, which contains the base URL for node-to-node HTTP services.
```

Reply:
the DID document contains a service with that type, not the service type